common/src/stack/command/stack/commands/add/vm/storage/__init__.py

```diff
@@ -0,0 +1,156 @@
+	or a path to a disk device (such as /dev/sdb) to mount to a virtual machine .
+	</param>
+
+	<example cmd='add vm storage virtual-backend-0-1 storage_pool=/export/pool/stacki disks=200'>
+	Add a new disk to virtual-backend-0-1 at the location of /export/pool/stacki/virtual-backend-0-1
+	Disks are always sorted into folders by virtual machine host name
+	</example>
+
+	<example cmd='add vm storage virtual-backend-0-1 storage_pool=/export/pool/stacki disks=/export/stacki.qcow2,/dev/sdb'>
+	Add two new disks to virtual-backend-0-1: the first is a premade qcow2 image which will be the boot disk
+	(order is from list of being added), the second is a disk that will be mounted to the VM as a mountpoint
+	</example>
+
+	"""
+	def valid_image(self, file_path, img_types):
+		"""
+		Use qemu-img to check the VM image
+		we are given is valid.
+		"""
+
+		valid_img = False
+
+		# Get the image info as json
+		output = _exec(f'qemu-img info --output=json {file_path}', shlexsplit=True)
+		if output.stderr:
+			return valid_img
+		img_info = json.loads(output.stdout)
+		image_ext = file_path.suffix.replace('.', '')
+
+		# Check qemu-img info and file extension
+		# as it classifies any file as raw
+		if img_info['format'] in img_types and image_ext in img_types:
+			valid_img = True
+		return valid_img
+
+	def run(self, params, args):
+		vm_host = self.getSingleHost(args)
+		vm_host = self.valid_vm_args([vm_host])
+
+		# Default naming scheme for newly made disk volumes
+		# is $HOSTNAME_disk followed by an incrementing number
+		# for each new disk
+		vol_name = f'{vm_host[0]}_disk'
+		dev_id = 0
+		vol_id = 0
+		disk_name = f'sd{chr(ord("a") + dev_id)}'
+		disk_loc, disks = self.fillParams([
+			('storage_pool', ''),
+			('disks', '', True)
+		])
+		if disk_loc:
+			disk_loc = Path(f'{disk_loc}/{vm_host[0]}')
+
+		# Valid image formats are qcow2 and raw
+		image_formats = ['qcow2', 'raw']
+
+		for disk in disks.split(','):
+			image = ''
+			disk_size = None
+			mount_disk = None
+
+			# Get the current disks
+			vm_disks = self.call('list.vm.storage', vm_host)
+
+			# Used for calculating the device name
+			disk_names = [ disk['Name'] for disk in vm_disks ]
+
+			# Used for disks created on the hypervisor
+			vol_names = [ disk['Image Name'] for disk in vm_disks if disk['Image Name'] and disk['Type'] == 'disk']
+			disk_path = Path(disk)
+
+			# If the disk is a number it's a disk to be created
+			# during vm definition
+			if disk.isdigit():
+				if not disk_loc:
+					raise ParamError(self, 'storage_pool', 'needed for defined disks or images')
+				disk_type = 'disk'
+				disk_size = disk
+				if vol_names:
+
+					# Get the next disk name but only consider other disk type images
+					vol_file = vol_names[-1].replace(Path(vol_names[-1]).suffix, '')
+					vol_id = int(vol_file.split(vol_name)[1]) + 1
+				else:
+					vol_id += 1
+				image = f'{vol_name}{vol_id}.qcow2'
+
+			elif '/dev/' in disk:
```

Comment:
See earlier conversation about 'startswith()'

Reply:
Looking over Ryan's suggestions I noticed this as well